### PR TITLE
Support `from_pretrain` for DIAR

### PR DESCRIPTION
**Pull request by YushiUeda**
This PR is related to https://github.com/espnet/espnet/pull/3535.
This is to support `from_pretrain` also for DIAR.

### CODE REVIEW

File: espnet2/bin/diar_inference.py [hunk 1]
```diff
@@ -159,6 +160,36 @@ def __call__(
...
+                from espnet_model_zoo.downloader import ModelDownloader
+
+            except ImportError:
+                logging.error(
+                    "`espnet_model_zoo` is not installed. "
+                    "Please install via `pip install -U espnet_model_zoo`."
+                )
+                raise
+            d = ModelDownloader()
+            kwargs.update(**d.download_and_unpack(model_tag))
```
Comment (kan-bayashi) [inline espnet2/bin/diar_inference.py:189]: The name of returned arguments are different.
https://github.com/espnet/espnet/blob/fb75313bd1ddf027e075ea7197448c929b176f5d/espnet2/bin/pack.py#L29-L32
Please change the names or add workaround solution.
https://github.com/kan-bayashi/espnet/blob/7f42a925b82dc722df7d93f08e07a1a62df390b3/espnet2/bin/enh_inference.py#L283-L285

File: espnet2/bin/diar_inference.py [hunk 2]
```diff
@@ -294,8 +330,16 @@ def get_parser():
...
+    group.add_argument(
+        "--diar_train_config",
+        type=str,
+        help="Diarization training configuration",
+    )
+    group.add_argument(
+        "--diar_model_file",
+        type=str,
+        help="Diarization model parameter file",
+    )
```
Comment (kan-bayashi) [inline espnet2/bin/diar_inference.py:342]: `model_tag` option is not added.

File: espnet2/bin/diar_inference.py (hunk 1 above)
Comment (YushiUeda) [reply espnet2/bin/diar_inference.py]: Thanks for the review. I corrected it as in `enh_inference.py`.

File: espnet2/bin/diar_inference.py (hunk 2 above)
Comment (YushiUeda) [reply espnet2/bin/diar_inference.py:344]: added `model_tag`

File: espnet2/bin/diar_inference.py (hunk 1 above)
Comment (kan-bayashi) [reply espnet2/bin/diar_inference.py]: Thanks!

**Pull request merged**